specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchservice.json

```diff
@@ -7657,6 +7657,38 @@
       },
       "description": "A skill to translate text from one language to another."
     },
+    "DocumentExtractionSkill": {
+      "x-ms-discriminator-value": "#Microsoft.Skills.Util.DocumentExtractionSkill",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SearchIndexerSkill"
+        }
+      ],
+      "properties": {
+        "parsingMode": {
+          "type": "string",
+          "x-nullable": true,
+          "description": "The parsingMode for the skill. Will be set to 'default' if not defined."
+        },
+        "dataToExtract": {
+          "type": "string",
+          "x-nullable": true,
+          "description": "The type of data to be extracted for the skill. Will be set to 'contentAndMetadata' if not defined."
+        },
+        "configuration": {
```

Comment:
Is there a way to provide a list of valid configuration options here? Or should we just leave it as flexible and let it fail at runtime if the customer specifies an invalid option?

Also, I want to make sure we define this correctly, as we had a bug with the SDK for the WebApiSkill (custom skill) when it was first released where it just straight didnt work with the SDK because the httpHeaders dictionary was configured incorrectly. Can we add a test to make sure this works as expected, just to be safe? Fixing that bug was technically a "breaking" change even though it never actually worked the first time, weirdly enough.

Reply:
Your best bet would be to use one of the code generators in your language of choice to generate the code locally.  What language would you want?  I can help find instructions if you don't know how to do that.

Reply:
It would be C#

Reply:
The various JSON under examples act as some kind of tests right?

As long as we define this the same way that WebApiHttpHeaders is defined now it should probably be fine. That is slightly different than this as is though (we created a separate object for WebApiHttpHeaders instead of defined it as a dictionary in line).

Reply:
I believe [this](https://github.com/Azure/autorest.csharp#use-outside-of-the-azure-sdk-net-repo) is what you want for generating C# code.  From there you'd write some app that ingests this code and run it.

@pakrym @ShivangiReja own the C# code generator and might have better instructions.

Reply:
cc @chamons 

The easiest way to try would be updating https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/autorest.md#L16-L17 to point to this revision and running `dotnet build /t:GenerateCode` in the src directory.

Reply:
cc @heaths

Reply:
Awesome.  Is there anything else remaining?

Reply:
I'll meet with the API review board tomorrow and hopefully we can merge the change after the meeting. Thanks!